stl/inc/xstring

```diff
@@ -2546,69 +2660,207 @@ public:
+            _My_data._Myres = _BUF_SIZE; // TRANSITION: constexpr SSO
+        }
+#endif // _HAS_CXX20
+
+        if constexpr (_Is_fwd_iter_v<_Iter>) {
+            const auto _Count = _Convert_size<size_type>(static_cast<size_t>(_STD distance(_First, _Last)));
+            if (_Count > max_size()) {
+                _Xlen_string(); // result too long
+            }
+
+#if _HAS_CXX20
+            const bool _Become_large = _Count >= _BUF_SIZE || _STD is_constant_evaluated();
+#else // ^^^ _HAS_CXX20 / !_HAS_CXX20 vvv
+            const bool _Become_large = _Count >= _BUF_SIZE;
+#endif // _HAS_CXX20
+            if (_Become_large) {
+                const size_type _New_capacity = _Calculate_growth(_Count);
+                const pointer _New_ptr        = _Al.allocate(_New_capacity + 1); // throws
+                _Construct_in_place(_My_data._Bx._Ptr, _New_ptr);
+                _My_data._Myres = _New_capacity;
+
+#if _HAS_CXX20
+                if (_STD is_constant_evaluated()) { // Begin the lifetimes of the objects before copying to avoid UB
+                    _Traits::assign(_Unfancy(_New_ptr), _New_capacity + 1, _Elem());
+                }
+#endif // _HAS_CXX20
+            }
+        }
+
+        _Tidy_deallocate_guard<basic_string> _Guard{this};
+        for (; _First != _Last; ++_First) {
+            if constexpr (!_Is_fwd_iter_v<_Iter>) {
+                if (_My_data._Mysize >= _My_data._Myres) { // Need to grow
+                    if (_My_data._Mysize > max_size()) {
+                        _Xlen_string(); // result too long
+                    }
+
+                    const auto _Old_ptr           = _My_data._Myptr();
+                    const size_type _New_capacity = _Calculate_growth(_My_data._Mysize);
+                    const pointer _New_ptr        = _Al.allocate(_New_capacity + 1); // throws
+
+#if _HAS_CXX20
+                    if (_STD is_constant_evaluated()) { // Begin the lifetimes of the objects before copying to avoid UB
+                        _Traits::assign(_Unfancy(_New_ptr), _New_capacity + 1, _Elem());
+                    }
+#endif // _HAS_CXX20
+                    _Traits::copy(_Unfancy(_New_ptr), _Old_ptr, _My_data._Mysize);
+                    if (_My_data._Myres >= _BUF_SIZE) { // Need to deallocate old storage
+                        _Al.deallocate(_My_data._Bx._Ptr, _My_data._Myres + 1);
+                        _My_data._Bx._Ptr = _New_ptr;
+                    } else {
+                        _Construct_in_place(_My_data._Bx._Ptr, _New_ptr);
+                    }
+                    _My_data._Myres = _New_capacity;
+                }
+            }
+
+            _Elem* const _Ptr = _Mypair._Myval2._Myptr();
+            _Traits::assign(_Ptr[_Mypair._Myval2._Mysize], *_First);
+            ++_Mypair._Myval2._Mysize;
+        }
+
+        _Elem* const _Ptr = _Mypair._Myval2._Myptr();
+        _Traits::assign(_Ptr[_Mypair._Myval2._Mysize], _Elem());

```

Comment:
This function should continue using `_My_data` for clarity (it isn't going to change).